backend/jack_backend.cpp

```diff
@@ -253,40 +328,74 @@ int JackMix::process( jack_nframes_t nframes, void* arg ) {
 	}
 	/// Adjust inlevels.
 	for ( in_it = backend->in_ports.begin(); in_it != backend->in_ports.end(); ++in_it ) {
-                QString key {in_it.key()};
-		jack_default_audio_sample_t* tmp = ins[key];
-		float volume = backend->getInVolume(key);
-                float max {0};
-		for ( jack_nframes_t n=0; n<nframes; n++ ) {
-                        tmp[ n ] *= volume;
-                        max = qMax(max, static_cast<float>(tmp[n]));
-                }
-                backend->newInputLevel(key, max);
+		QString key {in_it.key()};
+		jack_default_audio_sample_t* tmp = ins[ key ];
+		float volume = backend->getInVolume( key );
+		float volume_new = backend->getInVolumeNew( key );
+		float max {0};
+		if ( qFabs( ( volume_new - volume ) / volume ) > threshold) {
```

Comment:
Is threshold just a very small thing or have you thought about it?

I'm thinking maybe use https://doc.qt.io/qt-5/qtglobal.html#qFuzzyCompare instead?

Reply:
Threshold maybe should theoretically be the largest percent volume change in a single step that produces an acceptably small artifact.  I don't know what this value should be.  I put it there to be determined.  It kinda makes sense to me that there would be some threshold value to use.

Reply:
OK. I thought it was rounding thing (because you shouldn't do if a == b; a, b floating point).

I'm harbouring ideas about using a hanning cross-fade anyway in the next release. That'd be even better :)

Reply:
There is at least one place where I do if a == b; a, b floating point where fuzzy compare might be good, actually.